### ISSUE TITLE
Dialyzer error in Phoenix views when using Appsignal Elixir

### ISSUE THREAD

**Issue by ckruse** [appsignal/appsignal-elixir#588]
When adding the line `use Appsignal.Phoenix.View` in the view macro Dialyzer generates an error message for every Phoenix view in the project:

~~~
The guard test:

is_atom(
  _ ::
    binary()
    | maybe_improper_list(
        binary() | maybe_improper_list(any(), binary() | []) | char(),
        binary() | []
      )
)

can never succeed.
~~~

## To Reproduce

- Using AppSignal for Elixir package version 2.0.0
- In my app using Phoenix 1.5.6
- With this code:
  ```elixir
  def view do
    quote do
      use Phoenix.View,
        root: "lib/termitool_web/templates",
        namespace: TermitoolWeb

      use Appsignal.Phoenix.View
  end
  ```

**Comment by jeffkreeftmeijer** [comment]
Hey @ckruse, thanks for letting us know about this. I’ve pushed a fix which we’re currently [reviewing](https://github.com/appsignal/appsignal-elixir-phoenix/pull/6). If you have time, could you try it out locally by switching your app to using that branch, to see if that resolves the issue for you?

``` elixir
  defp deps do
    [
      {:appsignal_phoenix, github: "appsignal/appsignal-elixir-phoenix", branch: "non-binary-templates"},
    ]
  end
```

**Comment by ckruse** [comment]
hey @jeffkreeftmeijer, looks good - thanks :-)

**Comment by jeffkreeftmeijer** [comment]
I’ve just released version 2.0.1 of `:appsignal_phoenix` with this patch. To upgrade to it, you can switch back to the dependency you had before and run `mix deps.update appsignal_phoenix`:

```elixir
  defp deps do
    [
      {:appsignal_phoenix, “~> 2.0"},
    ]
  end
```
